src/jwt_authorizer/verify.py

```diff
@@ -0,0 +1,307 @@
+
+        Returns
+        -------
+        key : `bytes`
+            The issuer's key in PEM format.
+
+        Raises
+        ------
+        KeyClientException
+            Unable to retrieve the key set for the specified issuer.
+        UnknownAlgorithException
+            The requested key ID was found, but is for an unsupported
+            algorithm.
+        UnknownKeyIdException
+            The requested key ID was not present in the issuer configuration
+            or was not found in that issuer's JWKS.
+
+        Notes
+        -----
+        This function will automatically cache the last 16 keys for up to 10
+        minutes to cut down on network retrieval of the keys.
+        """
+        cache_key = hashkey(issuer, key_id)
+        if cache_key in self._cache:
+            return self._cache[cache_key]
+
+        self._logger.info("Getting key %s from %s", key_id, issuer.url)
+        if issuer.key_ids and key_id not in issuer.key_ids:
+            msg = f"kid {key_id} not found in configuration for {issuer.url}"
+            raise UnknownKeyIdException(msg)
+
+        # Retrieve the JWKS information.
+        keys = await self._key_client.get_keys(issuer)
+
+        # Find the key that we want.
+        for k in keys:
+            if key_id == k["kid"]:
+                key = k
+        if not key:
+            msg = f"Issuer {issuer.url} has no kid {key_id}"
+            raise UnknownKeyIdException(msg)
+        if key["alg"] != ALGORITHM:
+            msg = (
+                f"Issuer {issuer.url} kid {key_id} had algorithm"
+                f" {key['alg']} not {ALGORITHM}"
+            )
+            raise UnknownAlgorithmException(msg)
+
+        # Convert, cache, and return the key.
+        e = base64_to_number(key["e"])
+        m = base64_to_number(key["n"])
+        public_key = self._build_public_key(e, m)
+        self._cache[cache_key] = public_key
+        return public_key
+
+    @staticmethod
+    def _build_public_key(exponent: int, modulus: int) -> bytes:
+        """Convert an exponent and modulus to a PEM-encoded key."""
+        components = rsa.RSAPublicNumbers(exponent, modulus)
+        public_key = components.public_key(backend=default_backend())
+        return public_key.public_bytes(
+            encoding=Encoding.PEM, format=PublicFormat.SubjectPublicKeyInfo,
+        )
+
+
+def create_token_verifier(request: web.Request) -> TokenVerifier:
+    """Create a TokenVerifier from an app configuration.
+
+    Parameters
+    ----------
+    request : `aiohttp.web.Request`
+        The incoming request.
+
+    Returns
+    -------
+    token_verifier : `TokenVerifier`
+        A TokenVerifier created from that Flask application configuration.
+    """
+    logger: Logger = request["safir/logger"]
+    config: Config = request.config_dict["jwt_authorizer/config"]
+
+    if "jwt_authorizer/key_client" in request.config_dict:
+        key_client = request.config_dict["jwt_authorizer/key_client"]
```

Comment:
is this set somewhere else?  I'm trying to do some searching but I don't see this used elsewhere?

Reply:
Yeah, it's set in `create_app` if the `key_client` parameter is passed, which is done for the test suite. This is all cleaned up in my next PR.